_posts/2019-07-29-vim-basics.markdown

```diff
@@ -0,0 +1,111 @@
+
+Open terminal and type: `vim vim-basic`.
+
+You'll see something like the following:
+```
+                                              VIM - Vi IMproved
+
+                                               version 8.1.500
+                                           by Bram Moolenaar et al.
+                                 Vim is open source and freely distributable
+
+                                        Help poor children in Uganda!
+                                type  :help iccf<Enter>       for information
+
+                                type  :q<Enter>               to exit
+                                type  :help<Enter>  or  <F1>  for on-line help
+                                type  :help version8<Enter>   for version info
+
+                                        Running in Vi compatible mode
+                                type  :set nocp<Enter>        for Vim defaults
+                                type  :help cp-default<Enter> for info on this
+```
+
+congrats! you are using vim ha!
+
+
+## VIM modes
+Vim is a "modal editor", which means vim will behave different depending on which mode you are currently on.
+
+I'll introduce you to three basic modes. **NORMAL**, **INSERT** and **VISUAL**. There are a couple more, but these are the basics ones, and they are all you need to know for now.
+
+
+## NORMAL - The default vim mode
+
+This is where you'll spend most of your time and it's the mode where the magic happens. In this mode you can execute commands.
+Commands can be a simple letter, for example if you press the letter `j` you'll see your cursor goes down one line and if you press the letter `k` it will go up.
+In this mode you'll do things like navigate through your text file, removing, copying and pasting text.
+
+### Navigation
+While you are in **NORMAL** mode you would need to move around your text file. These are the basics commands that will help you achieve that.
+
+```
+      k
+      ^
+  h < + > l
+      v
+      j
+```
+
+### Removing text
+  dd - delete the current line.
+
+### Copy text
+  yy - copy(yank) the current line.
+
+### Pasting text
+  p  - paste to the line underneath.
+
+
+## INSERT - The mode where you can "insert" text
+  You probably were wondering, how do I actually write something into vim, well... this is the mode where you can do that.
+
+  You enter INSERT mode by pressing the letter `i`. In this mode you'll use your keyboard like in any other editor, vim will not behave in a special way when you are in this mode.
+
+  In vim, you are not expected to be in this mode all the time Whenever you finish your writing you go back to **NORMAL** mode, you do that by pressing the **ESC** key.
+
+  After writing or editing your changes you'll probably want to save your file and quit vim.
+  Press **ESC** and go back to normal mode, then you could do the any of following:
+
+  * :w  - Save your changes.
+  * :q  - Quit vim.
+  * :q! - Quit vim even if your changes were not saved.
```

Comment:
@bronzdoc It will be useful to mention `:wq` too

Reply:
🤔 `:wq` is the combination of two of the commands i'm showing